pkg/agent/agent.go

```diff
@@ -106,10 +116,14 @@ func (k *Klocksmith) process(stop <-chan struct{}) error {
 		return err
 	}
 
-	// we are schedulable now.
-	glog.Info("Marking node as schedulable")
-	if err := k8sutil.Unschedulable(k.nc, k.node, false); err != nil {
-		return err
+	if makeSchedulable {
+		// we are schedulable now.
+		glog.Info("Marking node as schedulable")
+		if err := k8sutil.Unschedulable(k.nc, k.node, false); err != nil {
+			return err
+		}
+	} else {
+		glog.Info("Skipping marking node as schedulable -- node was unschedulable prior to update reboot")
```

Comment:
Just a nit, but based on the case that we've discussed this isn't always the case. This could be discovered by a daemonset that is restarted but without an actual node reboot. Might be nice to just say something like "Skipping marking node as schedulable -- node was marked unschedulable by an external source" or something similar.